internal/pkg/auth/auth.go

```diff
@@ -59,42 +61,73 @@ func PersistLogoutToConfig(config *v1.Config) error {
 	return config.Save()
 }
 
-func PersistConfluentLoginToConfig(config *v1.Config, username, url, token, caCertPath string, isLegacyContext bool) error {
+func PersistConfluentLoginToConfig(config *v1.Config, credentials *Credentials, url, token, caCertPath string, isLegacyContext, save bool) error {
+	username := credentials.Username
 	state := &v1.ContextState{AuthToken: token}
 	var ctxName string
 	if isLegacyContext {
 		ctxName = GenerateContextName(username, url, "")
 	} else {
 		ctxName = GenerateContextName(username, url, caCertPath)
 	}
-	return addOrUpdateContext(config, ctxName, username, url, state, caCertPath, "")
+	return addOrUpdateContext(config, false, credentials, ctxName, url, state, caCertPath, "", save)
 }
 
-func PersistCCloudCredentialsToConfig(config *v1.Config, client *ccloudv1.Client, url string, credentials *Credentials) (*ccloudv1.Account, *ccloudv1.Organization, error) {
+func PersistCCloudCredentialsToConfig(config *v1.Config, client *ccloudv1.Client, url string, credentials *Credentials, save bool) (*ccloudv1.Account, *ccloudv1.Organization, error) {
 	ctxName := GenerateCloudContextName(credentials.Username, url)
 	user, err := client.Auth.User(context.Background())
 	if err != nil {
 		return nil, nil, err
 	}
 	state := getCCloudContextState(config, ctxName, credentials.AuthToken, credentials.AuthRefreshToken, user)
 
-	err = addOrUpdateContext(config, ctxName, credentials.Username, url, state, "", user.GetOrganization().GetResourceId())
+	err = addOrUpdateContext(config, true, credentials, ctxName, url, state, "", user.GetOrganization().GetResourceId(), save)
 	return state.Auth.Account, user.GetOrganization(), err
 }
 
-func addOrUpdateContext(config *v1.Config, ctxName, username, url string, state *v1.ContextState, caCertPath, orgResourceId string) error {
-	credName := generateCredentialName(username)
+func addOrUpdateContext(config *v1.Config, isCloud bool, credentials *Credentials, ctxName, url string, state *v1.ContextState, caCertPath, orgResourceId string, save bool) error {
+	credName := generateCredentialName(credentials.Username)
 	platform := &v1.Platform{
 		Name:       strings.TrimPrefix(url, "https://"),
 		Server:     url,
 		CaCertPath: caCertPath,
 	}
 	credential := &v1.Credential{
 		Name:     credName,
-		Username: username,
+		Username: credentials.Username,
 		// don't save password if they entered it interactively.
 	}
 
+	if save && !credentials.IsSSO {
+		salt, err := secret.GenerateRandomBytes(24)
+		if err != nil {
+			return err
+		}
+		nonce, err := secret.GenerateRandomBytes(12)
+		if err != nil {
+			return err
+		}
+
+		encryptedPassword, err := secret.Encrypt(credentials.Username, credentials.Password, salt, nonce)
+		if err != nil {
+			return err
+		}
+
+		loginCredential := &v1.LoginCredential{
+			IsCloud:    isCloud,
+			IgnoreCert: !isCloud,
```

Comment:
Is this value always equal to `!isCloud`?

Reply:
yes

Reply:
it's set to false for cloud login and true for confluent login

Reply:
so for saving login credential purposes, it's always `!isCloud`